enhancements/image-references.md

```diff
@@ -0,0 +1,225 @@
+  which any individual build system will resolve image references
+- upstream operator manifests must not be required to know how common names or
+  referecnes change when built in a downstream pipeline
+- no prescriptive statement is made about the specific format or contents of the
+  bundle image layer; any will be transparently supported as an output
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an author of a manifest, I would like to check in manifests to my upstream
+repository that are self-consistent, valid and make no assumptions about the
+build system that will eventually create a bundle image with them.
+
+#### Story 2
+
+As an author of an operand, I would like to create a bundle locally in order
+to test my operator end-to-end on a cluster of my choosing without having to
+edit the core configuration for the operator.
+
+#### Story 3
+
+As an author of a build system, I would like to operate with tooling that allows
+me to clearly define the source of truth for image digests in order to keep the
+build-system-specific configuration to an absolute minimum.
+
+#### Story 4
+
+As an engineer involved in publishing an optional operator, I would like to
+configure semantically equivalent image pull specifications once, in order to not
+need to configure each build system independently.
+
+### Implementation Details/Notes/Constraints
+
+The core problems that must be solved in the implementation of this proposal have
+already been handled in the workflow used in `oc adm release new`. When implementing
+improvements to the `operator-sdk bundle create` process we will simply need to
+create a shared library for the two tools to use. While the shape of the output
+is slightly different and some of the semantics about how the output should be 
+formatted are dissmilar, the core image reference rewriting is identical and the
+process of building a `FROM scratch` image layer is also identical.
+
+Today, some prior art exists in the OSBS workflow for building operator bundles.
+As we improve the Operator SDK tooling to create a straghtforward process for
+creating bundle images, we must make sure a seamless migration is possible.
+
+### Risks and Mitigations
+
+It will be critical that the design be vetted by all of the concerned parties, from
+operator manifest authors to CI system authors and productized pipeline authors to
+ensure that the UX is appropriate in all cases. Furthermore, the largest risk in the
+implementation here is not prioritizing a clean migration pathway for all current
+users who create bundle images, which would lead to further fragmentation of the
+ecosystem, which is directly opposed to the goal of this enhancement.
+
+## Design Details
+
+The definition of a minimally-viable operator bundle image will be changed to
+ensure that all iamge references in the contained manifests have been resolved 
+to a digest and had the pull specifications rewritten to refer to those digests.
```

Comment:
I think this constraint should be opt-in for upstream users and mandatory for admittance to catalogs for k8s distros that require it.

Reply:
I disagree. A bundle with unresolved image references is not reproducible and will be hard to support in any context it's used. If the bundle creation tooling makes it trivial to resolve a spec to a digest, what use would someone have for an unresolved spec in a bundle?

Reply:
@stevekuznetsov to clarify: are you saying that you want all bundles to include `image-references` and `image-replacements`?

Reply:
I'm not sure I see the argument here - how are `image-references` and `image-replacements` useful for users that don't have the split build systems / pipelines that openshift has?

An upstream user would build the operator/operand images, record their digests in their bundle manifests, and publish it.

Reply:
If an upstream user still wants to create reproducible artifacts they still want to SHA-pin. I agree at that point though image digest resolution could be a simple "look up the current digest for this floating tag" without an indirection, so the `image-replacements` would be inferred.